### PR TITLE
Add stacked `KUBECONFIG` support

### LINKED ISSUE #132: Stacked kubeconfigs don't work
The Go client (including the `kubectl` tool) supports merging multiple kubeconfigs, as described at https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#merging-kubeconfig-files.

# Repro steps

1. Check out 9741360ce0fb71bba0bd2cd0d133c2d530cc573e (also happens with the 0.25.0 release)
2. Create two empty stacked kubeconfigs: `touch /tmp/{a,b}`
3. Run the `node_reflector` example: `env KUBECONFIG=/tmp/a:/tmp/b cargo run --features openapi --example node_reflector`

# Expected behaviour

`/tmp/a` and `/tmp/b` are loaded and merged. `node_reflector` crashes because the merged kubeconfig still doesn't say which cluster to connect to.

# Actual behaviour

`node_reflector` tries to load the literal path `/tmp/a:/tmp/b`, and crashes because it does not exist.

# Workaround

Use `kubectl` to flatten the kubeconfig: `kubectl config --flatten > tmp-kubeconfig && env KUBECONFIG=tmp-kubeconfig cargo run ...`.

### PR DESCRIPTION
**Pull request by kazk**
- Cleaned up `config::util`
- `current_context` in `KubeConfig` is now `Option<String>`
- File paths in `Kubeconfig` are now remapped to absolute path after reading to support relative paths in stacked config. Assumes paths are unicode (should be fine because Go doesn't have OsString concept).
- Kubeconfigs are fully merged according to <https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#merging-kubeconfig-files> even if `kube` only needs `current_context`, `contexts`, `clusters`, and `auth_infos`. If we don't need them, we should remove other fields from `Kubeconfig` as well because it'll be misleading. I think it's fine to keep.
  - For `Option<Preferences>` and `Option<Vec<NamedExtension>>`, we're just taking the first `Some` because it's unclear what should be done.
- Returns error when `apiVersion`/`kind` is set and mismatches. Not sure if this is correct behavior.

Closes #132

### CODE REVIEW

File: kube/src/config/file_config.rs
Comment (clux) [inline kube/src/config/file_config.rs:274]: This feels like the type of thing where `extensions` should be ` Vec<String>` with `[serde(default, skip_serialize_if = "Option::is_none")]` rather than `Option<Vec<String>>` since we treat missing as empty anyway.

Though it seems we have a convention for `Option<Vec<.>>` atm so it's a bit of a breaking change :thinking:

File: kube/src/config/file_config.rs
Comment (clux) [reply kube/src/config/file_config.rs:197]: Think we can make a `osPathSeparator` const inside a `#[cfg(target_os = "windows")]`

**Review by clux — COMMENTED** [review]
delimit comment + nits

looking great. you are really cleaning up the place.

File: kube/src/config/file_config.rs
Comment (kazk) [reply kube/src/config/file_config.rs]: I just did `.split(if cfg!(windows) { ';' } else { ':' })` since we don't need it anywhere else.

**Review by clux — APPROVED** [review]

File: kube/src/config/file_config.rs
Comment (kazk) [reply kube/src/config/file_config.rs:238]: Oh, it can just be
```rust
self.kind = self.kind.or(next.kind);
```
:laughing: I don't know why I didn't think of that.

`.ok_or_else` needs `.ok`
```rust
self.kind = self.kind.ok_or_else(|| next.kind.take()).ok();
```

File: kube/src/config/file_config.rs
Comment (kazk) [inline kube/src/config/file_config.rs:247]: Maybe we shouldn't merge this? ("Never change the value or map key.")

Changes `preferences.extensions` value. Just update if `None`?

File: kube/src/config/file_config.rs
Comment (teozkr) [reply kube/src/config/file_config.rs]: FWIW, Rust has a built in function for this: https://doc.rust-lang.org/stable/std/env/fn.split_paths.html

File: kube/src/config/file_config.rs
Comment (clux) [reply kube/src/config/file_config.rs:247]: Hm, really not sure about this. Don't know enough about whether or not adding extensions together makes sense.
It's probably not super important at the moment since it's more a kubectl option afaikt, and we are not doing anything with it. I'd just take the first preferences that's a Some and not worry about it for now.

File: kube/src/config/file_config.rs
Comment (kazk) [reply kube/src/config/file_config.rs]: Thanks! Updated to use that and `std::env::var_os`.

**Comment by kazk** [comment]
I think it's good enough now. I'll merge after cleaning up the commits.

**Review by clux — APPROVED** [review]
looking great. got tests, super readable now.
there's enough to make a new release with this, are you at a good point for 0.50.0?

**Comment by kazk** [comment]
Yeah :+1: I don't think there's anything else I'd like to include in the next release and want to unblock users with this fix.

**Comment by kazk** [comment]
> you are really cleaning up the place.

@clux Thank you for being receptive and quick to respond. It's been a pleasure to work on `kube`.

I know I went a little crazy the past week :) `hyper` + `tower` got me excited, and I wanted to do some clean up while I was off from work.
I'm back to work now, but will start implementing a project using `kube` (client and controller). Maybe I'll explore unit testing approaches for clients. Now with `tower`, it should be possible to create a fake like in `client-go` by creating a mock from resources and using that to create a client. Maybe `kube-test` crate can provide some convenience builder for it.

**Comment by clux** [comment]
It's been amazing. The huge hyper/tower hotswap, connection upgrading, tls rewrites, config rewrites, plus all the other hard stuff you've tackled like websockets subresources and schemas.

Really appreciate all the work you've done here. It's made the library a lot more usable and the rust kubernetes story a lot more attractive (the crd reception was huge). Looking forward to hearing what issues you uncover on the controller side. I am back at work now as well, but reviewing your prs has been the easiest thing to do on the side, always happy to look at more :-)

**Pull request merged**
